document/js-api/index.bs

```diff
@@ -761,29 +761,39 @@ which can be simultaneously referenced by multiple {{Instance}} objects. Each
 </div>
 
 <div algorithm>
-    The <dfn constructor for="Table">Table(|descriptor|)</dfn> constructor, when invoked, performs the following steps:
+    The <dfn constructor for="Table">Table(|descriptor|, |value|)</dfn> constructor, when invoked, performs the following steps:
+    1. Let |elementType| be |descriptor|["element"].
     1. let |initial| be |descriptor|["initial"].
     1. If |descriptor|["maximum"] is [=present=], let |maximum| be |descriptor|["maximum"]; otherwise, let |maximum| be empty.
     1. If |maximum| is not empty and |maximum| &lt; |initial|, throw a {{RangeError}} exception.
-    1. Let |type| be the [=table type=] {[=table type|𝗆𝗂𝗇=] n, [=table type|𝗆𝖺𝗑=] |maximum|} [=table type|funcref=].
+    1. If |value| is missing and |elementType| is {{TableKind/anyfunc}},
```

Comment:
What happens for other types? Isn't the next step ill-defined then?
Either way, I would simply make the default to null independent of the element type, seems cleanest.

Reply:
Agreed, this didn't end up quite right. However, it's not exactly clear what "right" means in the context of `optional any`, unfortunately (https://www.w3.org/Bugs/Public/show_bug.cgi?id=23602).

We probably want to support creating an `anyref` table with the values initialized to `undefined`. If we defaulted to `null` in that case, we'd either break that or make `new Table({..}, undefined)` do something different from `new Table({..})`. That's pretty surprising behavior in JS, so I don't think we should do that.

That leaves the `nullref` type to be considered; it looks like I proposed making `new Table({..}, undefined)` and `new Table({..})` both throw there. That's probably not so helpful.

I notice that the `Global` constructor has the same issues… I think the best solution will be to use `DefaultValue()`.